### ISSUE TITLE
linter: explicit-function-return-type being overzealous compared to eslint-typescript?

### ISSUE THREAD

**Issue by KieranP** [oxc-project/oxc#12704]
### What version of Oxlint are you using?

1.9.0

### What command did you run?

oxlint

### What does your `.oxlintrc.json` config file look like?

```jsonc
{
  "$schema": "./node_modules/oxlint/configuration_schema.json",
  "ignorePatterns": ["/node_modules", "/dist", "/*.{js,ts}"],
  "plugins": ["eslint", "oxc", "typescript", "import", "promise", "unicorn"],
  "categories": {
    "correctness": "error",
    "nursery": "error",
    "pedantic": "error",
    "perf": "error",
    "restriction": "error",
    "style": "error",
    "suspicious": "error"
  }
}
```

### What happened?

This is not so much an issue as it is a question regarding how compatible you want to be with the eslint versions of typescript checks.

When I enable `typescript/explicit-function-return-type` in oxlint, it picks up many more functions than eslint-typescript does. For example, I have this bit of code:

```ts
return clients.filter(
  (client) =>
    searchWords.every((word) =>
      client.name.toLowerCase().includes(word)
    ) ||
    client.cats.some((cat) =>
      searchWords.every((word) => cat.name.toLowerCase().includes(word))
    )
)
```

eslint-typescript's `typescript/explicit-function-return-type` does not report anything on this bit of code.

But oxlint's `typescript/explicit-function-return-type` check reports the following:

```
  × typescript-eslint(explicit-function-return-type): Missing return type on function.
    ╭─[src/lib/components/ClientList.svelte:24:18]
 23 │       return clients.filter(
 24 │         (client) =>
    ·                  ──
 25 │           client.primaryPhone?.toString().includes(searchTerm) ?? false
    ╰────
  help: Require explicit return types on functions and class methods.

  × typescript-eslint(explicit-function-return-type): Missing return type on function.
    ╭─[src/lib/components/ClientList.svelte:31:18]
 30 │       return clients.filter(
 31 │         (client) =>
    ·                  ──
 32 │           searchWords.every((word) =>
    ╰────
  help: Require explicit return types on functions and class methods.

  × typescript-eslint(explicit-function-return-type): Missing return type on function.
    ╭─[src/lib/components/ClientList.svelte:32:36]
 31 │         (client) =>
 32 │           searchWords.every((word) =>
    ·                                    ──
 33 │             client.name.toLowerCase().includes(word)
    ╰────
  help: Require explicit return types on functions and class methods.

  × typescript-eslint(explicit-function-return-type): Missing return type on function.
    ╭─[src/lib/components/ClientList.svelte:35:34]
 34 │           ) ||
 35 │           client.cats.some((cat) =>
    ·                                  ──
 36 │             searchWords.every((word) => cat.name.toLowerCase().includes(word))
    ╰────
  help: Require explicit return types on functions and class methods.

  × typescript-eslint(explicit-function-return-type): Missing return type on function.
    ╭─[src/lib/components/ClientList.svelte:36:38]
 35 │           client.cats.some((cat) =>
 36 │             searchWords.every((word) => cat.name.toLowerCase().includes(word))
    ·                                      ──
 37 │           )
    ╰────
  help: Require explicit return types on functions and class methods.
```

To fix oxlint, I have to add 'boolean' return type to all of the arrow functions:

```ts
return clients.filter(
  (client): boolean =>
    searchWords.every((word): boolean =>
      client.name.toLowerCase().includes(word)
    ) ||
    client.cats.some((cat): boolean =>
      searchWords.every((word): boolean =>
        cat.name.toLowerCase().includes(word)
      )
    )
)
```

Technically, exlint is correct in reporting them, functions that don't have return types. But perhaps eslint-typescript is excluding very simple arrow functions like this from needing to be typed?

How close do you want oxlint versions of the checks to be compared to eslint versions? Would this be considered a porting bug, or are the extra errors because of a bug in the eslint check?
